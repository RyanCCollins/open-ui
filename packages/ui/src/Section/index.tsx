import * as React from 'react';
import Component from './styles';
import BoxProps from '../Box/types';

export default function Section({
  children,
  ...rest,
}: BoxProps): JSX.Element {
  return (
    <Component {...rest}>
      {children}
    </Component>
  );
};
