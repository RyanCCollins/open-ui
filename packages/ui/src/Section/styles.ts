const styled = require('styled-components').default;
import {
  BoxStyles,
} from '../Box/styles';

export default styled.section`
  ${BoxStyles}
  width: 100%;
`;
