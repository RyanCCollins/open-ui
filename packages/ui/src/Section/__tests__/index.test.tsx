import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Section from '../';
import props from './__mocks__/sectionMocks.mock';

describe('<Section />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Section>
        <div />
      </Section>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with different props', () => {
    const wrapper = shallow(
      <Section {...props}>
        <div />
      </Section>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
