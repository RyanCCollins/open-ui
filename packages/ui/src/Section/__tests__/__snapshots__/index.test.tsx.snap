// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render with default props 1`] = `
<styled.section>
  <div />
</styled.section>
`;

exports[`<Section /> should render with different props 1`] = `
<styled.section
  backgroundColor="#f5f5f5"
  pad="medium"
  size="large"
>
  <div />
</styled.section>
`;
